core/common/src/main/java/alluxio/underfs/options/UnderFileSystemMkdirsOptions.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.underfs.options;
+
+import alluxio.Configuration;
+import alluxio.annotation.PublicApi;
+import alluxio.security.authorization.PermissionStatus;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * Method options for mkdirs in UnderFileSystem.
+ */
+@PublicApi
+@NotThreadSafe
+public final class UnderFileSystemMkdirsOptions {
+  // Permission status to set for the directories being created.
+  private PermissionStatus mPermissionStatus;
+  // Determine whether to create any necessary but nonexistent parent directories.
+  private boolean mCreateParent;
+
+  /**
+   * @return the default {@link UnderFileSystemMkdirsOptions}
+   */
+  public static UnderFileSystemMkdirsOptions defaults() {
```

Comment:
it seems odd to provide defaults() and a constructor ... it is mixing two idioms

we should either have two constructors or two defaults (defaults() and defaults(conf), which is what I prefer)